### PR TITLE
Add feature to simplify code to use the new extended property patterns when available.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Looks like this:

image

Todo:

- [x] tests

Fixes https://github.com/dotnet/roslyn/issues/55545

Relates to test plan https://github.com/dotnet/roslyn/issues/52468

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/SimplifyPropertyPattern/SimplifyPropertyPatternTests.cs
Comment (jcouv) [inline src/EditorFeatures/CSharpTest/SimplifyPropertyPattern/SimplifyPropertyPatternTests.cs:565]: Consider also testing a FixAll scenario

Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/SimplifyPropertyPattern/SimplifyPropertyPatternTests.cs:565]: i do :)

Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/SimplifyPropertyPattern/SimplifyPropertyPatternTests.cs:612]: Ooops.  Ignore me.  i hadn't pushed :)

File: src/Analyzers/CSharp/Analyzers/SimplifyPropertyPattern/CSharpSimplifyPropertyPatternDiagnosticAnalyzer.cs
Comment (jcouv) [inline src/Analyzers/CSharp/Analyzers/SimplifyPropertyPattern/CSharpSimplifyPropertyPatternDiagnosticAnalyzer.cs:28]: This seems wrong

Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/Analyzers/SimplifyPropertyPattern/CSharpSimplifyPropertyPatternDiagnosticAnalyzer.cs:28]: indeed!

File: src/Analyzers/CSharp/Analyzers/SimplifyPropertyPattern/CSharpSimplifyPropertyPatternDiagnosticAnalyzer.cs
Comment (jcouv) [inline src/Analyzers/CSharp/Analyzers/SimplifyPropertyPattern/CSharpSimplifyPropertyPatternDiagnosticAnalyzer.cs:37]: Would `SyntaxTreeWithoutSemanticsAnalysis` suffice?

File: src/Analyzers/CSharp/Analyzers/SimplifyPropertyPattern/SimplifyPropertyPatternHelpers.cs
Comment (jcouv) [inline src/Analyzers/CSharp/Analyzers/SimplifyPropertyPattern/SimplifyPropertyPatternHelpers.cs:28]: I saw tests with designation part, but not with type or positional part. Consider covering that too

File: src/EditorFeatures/CSharpTest/SimplifyPropertyPattern/SimplifyPropertyPatternTests.cs
Comment (jcouv) [inline src/EditorFeatures/CSharpTest/SimplifyPropertyPattern/SimplifyPropertyPatternTests.cs:60]: Can the fixer be invoked either on `Assembly:` or `EntryPoint:` in this case (yielding different results)?

File: src/Analyzers/CSharp/Analyzers/SimplifyPropertyPattern/SimplifyPropertyPatternHelpers.cs
Comment (jcouv) [inline src/Analyzers/CSharp/Analyzers/SimplifyPropertyPattern/SimplifyPropertyPatternHelpers.cs:49]: nit: Do we have a test where the expression is not a `MemberAccessExpressionSyntax`?

File: src/Analyzers/CSharp/Analyzers/SimplifyPropertyPattern/CSharpSimplifyPropertyPatternDiagnosticAnalyzer.cs
Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/Analyzers/SimplifyPropertyPattern/CSharpSimplifyPropertyPatternDiagnosticAnalyzer.cs:37]: unfortunately no:

```
 /// Note: an <see cref="DiagnosticAnalyzer"/> that uses this will not work properly if
/// it registers a <see cref="AnalysisContext.RegisterSyntaxNodeAction{TLanguageKindEnum}(Action{SyntaxNodeAnalysisContext}, TLanguageKindEnum[])"/> and then ends
```

File: src/Analyzers/CSharp/CodeFixes/SimplifyPropertyPattern/CSharpSimplifyPropertyPatternCodeFixProvider.cs
Comment (jcouv) [inline src/Analyzers/CSharp/CodeFixes/SimplifyPropertyPattern/CSharpSimplifyPropertyPatternCodeFixProvider.cs:73]: static?

File: src/Analyzers/CSharp/Analyzers/SimplifyPropertyPattern/CSharpSimplifyPropertyPatternDiagnosticAnalyzer.cs
Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/Analyzers/SimplifyPropertyPattern/CSharpSimplifyPropertyPatternDiagnosticAnalyzer.cs:37]: you only use that if you do RegisterSyntaxTreeAction

**Review by jcouv — COMMENTED** [review]
Done with review pass (iteration 6). Looks great, thanks!

**Review by jmarolf — APPROVED** [review]

File: src/Analyzers/CSharp/CodeFixes/SimplifyPropertyPattern/CSharpSimplifyPropertyPatternCodeFixProvider.cs
Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/CodeFixes/SimplifyPropertyPattern/CSharpSimplifyPropertyPatternCodeFixProvider.cs]: done.

File: src/EditorFeatures/CSharpTest/SimplifyPropertyPattern/SimplifyPropertyPatternTests.cs
Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/SimplifyPropertyPattern/SimplifyPropertyPatternTests.cs:60]: yup!

File: src/Analyzers/CSharp/Analyzers/SimplifyPropertyPattern/SimplifyPropertyPatternHelpers.cs
Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/Analyzers/SimplifyPropertyPattern/SimplifyPropertyPatternHelpers.cs:29]: will do!

**Comment by jinujoseph** [comment]
@CyrusNajmabadi this needs to retargeted to release/17.1 branch

**Comment by CyrusNajmabadi** [comment]
@jinujoseph Retargeted.  Thanks :)

**Comment by jcouv** [comment]
There seems to be test failures

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)

**Review by 333fred — COMMENTED** [review]
Looks like there's a merge of commits that aren't yet in main, will hold off approving until those commits are merged and this PR shows a clean diff.

**Comment by 333fred** [comment]
@CyrusNajmabadi it doesn't look like there was 2 compiler reviews. I was holding off on approving this until I could actually review a clean diff with the target branch, which is now impossible.

**Comment by CyrusNajmabadi** [comment]
This shouldnt' have needed 2 compiler reviews afaict.  All i did was merge main into this branch.  Do you review those changes?

**Comment by 333fred** [comment]
Any non-test changes that touch the compiler folder need 2 reviews. I cannot separate your changes from the merge, so no I was not able to review them.

**Comment by 333fred** [comment]
Moreover, because this pr was merged bringing in a whole bunch of unrelated commits, it means we can't easily revert the changes if necessary. In the future, _please_ do not merge a pr with unrelated commits in the merge because codeflow has not yet occurred.

**Comment by CyrusNajmabadi** [comment]
Ok, i won't do in the future.  This was done here though to unblock CI, which needed those changes to properly use sdk-preview7 and which had already been approved in https://github.com/dotnet/roslyn/pull/55681.  It pulled in 'main'.  But 'main' is already something we auto-approve for merge into later-release branches. 

Given the contents were all approved already, i didn't know there needed to be additional approval.  But in the future i can avoid this if it is problematic.

**Comment by 333fred** [comment]
>  It pulled in 'main'. But 'main' is already something we auto-approve for merge into later-release branches.

Right, I understand this. However, `main`, was not yet merged into this branch. This means that your PR effectively combined your code and https://github.com/dotnet/roslyn/pull/55756 into a single merge commit, which makes it very difficult to separate your changes from the main changes. I'm not saying you shouldn't have merged `main` into your branch to allow CI to run: I'm saying you should have waited for https://github.com/dotnet/roslyn/pull/55756 to merge before merging this, so that this PR would be _just_ your changes, and not your changes + everything in `main` since the last time it was merged into release/dev17.1-preview1.

**Pull request merged**
